test/pylib/util.py

```diff
@@ -74,7 +78,11 @@ async def get_hosts() -> Optional[list[Host]]:
 
         logging.info(f"Driver hasn't yet learned about hosts: {remaining}")
         return None
-    hosts = await wait_for(get_hosts, deadline)
+    hosts = await wait_for(
+        pred=get_hosts,
+        deadline=deadline,
+        before_retry=lambda: cql.cluster.refresh_nodes(force_token_rebuild=True),
```

Comment:
There is no motivation of this change in the commit message or PR cover letter.
Does anyone know why this was done? Did any of the reviewers look at this change and ask why it's necessary? @avikivity @nyh 
It is now causing failures in tests: https://github.com/scylladb/scylladb/issues/17616
and I don't understand why it was added in the first place, whether I can remove it.

Reply:
I checked if removing it causes the new test to fail, and indeed it does. So it looks like the new test somehow requires it.